bundle

```diff
@@ -128,29 +140,33 @@ def version_bundle(bundle, channel, overrides):
                 copy['applications'][application]['charm'] = name + '-' + revision
                 break
         else:
-            resp = requests.get('https://api.jujucharms.com/charmstore/v5/meta/id',
-                                params={'id': charm, 'channel': channel})
-            if not resp.ok:
-                err = "Couldn't query %s %s, is it public?" % (charm, channel)
-                raise CharmstoreQueryError(err)
-            meta = resp.json()
-            meta = [meta[k] for k in meta][0]
-            copy['applications'][application]['charm'] = meta['Id']
+            if charm.startswith("cs:"):
+                resp = requests.get('https://api.jujucharms.com/charmstore/v5/meta/id',
+                                    params={'id': charm, 'channel': channel})
+                if not resp.ok:
+                    err = "Couldn't query %s %s, is it public?" % (charm, channel)
+                    raise CharmstoreQueryError(err)
+                meta = resp.json()
+                meta = [meta[k] for k in meta][0]
+                copy['applications'][application]['charm'] = meta['Id']
+            else:
+                copy['applications'][application]['channel'] = channel
```

Comment:
each ~containers owned charm in charmhub (ie. `1.24/edge`) will be up-to-date based on its channel.  Therefore there's no need to use specific revisions